src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -51,6 +53,7 @@ object SnowflakeTelemetry {
   def sendClientInfoTelemetryIfNotYet(extraValues: Map[String, String],
                                       conn: Connection): Unit = {
     if (!hasClientInfoSent) {
+      SparkConnectorContext.recordConfig()
```

Comment:
What is the difference between recordConfig and sendClientInfoTelemetry

Reply:
recordConfig() is supposed to be called for both executor and driver to log configuration.
sendClientInfoTelemetry() is supposed to be called once on spark driver to send a spark_client_info telemetry message.

For spark driver, recordConfig() can be called in other places, but I think it is a good place to call it in sendClientInfoTelemetry().